diff -Naur gjs/meson.build gjs-clean/meson.build
--- gjs/meson.build	2026-08-09 09:52:01.030586388 +0300
+++ gjs-clean/meson.build	2026-08-09 10:35:16.807778266 +0300
@@ -282,36 +282,7 @@
 --disable-debug.''')
 endif
 
-# Check if a minimal SpiderMonkey program compiles, links, and runs. If not,
-# it's most likely the case that SpiderMonkey was configured incorrectly, for
-# example by building mozglue as a shared library.
-minimal_program = cxx.run('''
-#include <js/Initialization.h>
-int main() {
-    if (!JS_Init()) return 1;
-    JS_ShutDown();
-    return 0;
-}
-''',
-    args: debug_arg, dependencies: spidermonkey,
-    name: 'SpiderMonkey sanity check')
-
-recommended_configuration = ''' Check the recommended configuration:
-https://github.com/spidermonkey-embedders/spidermonkey-embedding-examples/blob/esr91/docs/Building%20SpiderMonkey.md'''
-if not minimal_program.compiled()
-    error('''A minimal SpiderMonkey program
-could not be compiled or linked. Most likely you should build it with a
-different configuration.''' + recommended_configuration)
-elif meson.is_cross_build()
-    warning('''This is a cross build. A check that a minimal
-SpiderMonkey program executes will not be performed. Before shipping GJS, you
-should check that it does not crash on startup, since building SpiderMonkey with
-the wrong configuration may cause that.''' + recommended_configuration)
-elif minimal_program.returncode() != 0
-    error('''A minimal SpiderMonkey program
-failed to execute. Most likely you should build it with a different
-configuration.''' + recommended_configuration)
-endif
+# meow
 
 ### Check for external programs ################################################
 

diff --git a/gjs/engine.cpp b/gjs/engine.cpp
--- a/gjs/engine.cpp
+++ b/gjs/engine.cpp
@@ -197,10 +197,29 @@
 JSContext* gjs_create_js_context(GjsContextPrivate* uninitialized_gjs) {
     g_assert(gjs_is_inited);
     JSContext* cx = JS_NewContext(/* max bytes = */ 32 * 1024 * 1024);
-    if (!cx)
+    if (!cx) {
+        g_printerr("gjs: JS_NewContext() failed\n");
         return nullptr;
+    }
+
+    bool enable_jit = !(g_getenv("GJS_DISABLE_JIT"));
+    if (enable_jit) {
+        gjs_debug(GJS_DEBUG_CONTEXT, "Enabling JIT");
+    }
+    JS::ContextOptionsRef(cx).setAsmJS(enable_jit);
+
+    uint32_t value = enable_jit ? 1 : 0;
+
+    JS_SetGlobalJitCompilerOption(
+        cx, JSJitCompilerOption::JSJITCOMPILER_ION_ENABLE, value);
+    JS_SetGlobalJitCompilerOption(
+        cx, JSJitCompilerOption::JSJITCOMPILER_BASELINE_ENABLE, value);
+    JS_SetGlobalJitCompilerOption(
+        cx, JSJitCompilerOption::JSJITCOMPILER_BASELINE_INTERPRETER_ENABLE,
+        value);
 
     if (!JS::InitSelfHostedCode(cx)) {
+        g_printerr("gjs: JS::InitSelfHostedCode() failed\n");
         JS_DestroyContext(cx);
         return nullptr;
     }
@@ -239,21 +258,5 @@
             "extra warnings.");
     }
 
-    bool enable_jit = !(g_getenv("GJS_DISABLE_JIT"));
-    if (enable_jit) {
-        gjs_debug(GJS_DEBUG_CONTEXT, "Enabling JIT");
-    }
-    JS::ContextOptionsRef(cx).setAsmJS(enable_jit);
-
-    uint32_t value = enable_jit ? 1 : 0;
-
-    JS_SetGlobalJitCompilerOption(
-        cx, JSJitCompilerOption::JSJITCOMPILER_ION_ENABLE, value);
-    JS_SetGlobalJitCompilerOption(
-        cx, JSJitCompilerOption::JSJITCOMPILER_BASELINE_ENABLE, value);
-    JS_SetGlobalJitCompilerOption(
-        cx, JSJitCompilerOption::JSJITCOMPILER_BASELINE_INTERPRETER_ENABLE,
-        value);
-
     return cx;
 }
