diff -Naur gnome-settings-daemon/meson.build gnome-settings-daemon-clean/meson.build
--- gnome-settings-daemon/meson.build	2022-09-18 00:37:19.000000000 +0300
+++ gnome-settings-daemon-clean/meson.build	2026-08-09 15:08:52.950121854 +0300
@@ -104,7 +104,7 @@
 libcanberra_gtk_dep = dependency('libcanberra-gtk3')
 libgeoclue_dep = dependency('libgeoclue-2.0', version: '>= 2.3.1')
 libnotify_dep = dependency('libnotify', version: '>= 0.7.3')
-libpulse_mainloop_glib_dep = dependency('libpulse-mainloop-glib', version: '>= 2.0')
+libpulse_mainloop_glib_dep = dependency('libpulse-mainloop-glib')
 pango_dep = dependency('pango', version: '>= 1.20.0')
 polkit_gobject_dep = dependency('polkit-gobject-1', version: '>= 0.114')
 upower_glib_dep = dependency('upower-glib', version: '>= 0.99.12')
