diff -Naur gtk4/gdk/loaders/gdkpng.c gtk4-clean/gdk/loaders/gdkpng.c
--- gtk4/gdk/loaders/gdkpng.c	2026-04-19 02:33:20.000000000 +0300
+++ gtk4-clean/gdk/loaders/gdkpng.c	2026-05-29 12:14:53.696240327 +0300
@@ -345,7 +345,7 @@
   png_set_read_user_chunk_fn (png, &cicp, png_read_chunk_func);
 #endif
 
-  if (sigsetjmp (png_jmpbuf (png), 1))
+  if (setjmp (png_jmpbuf (png)))
     {
       g_free (buffer);
       g_free (row_pointers);
@@ -664,7 +664,7 @@
   gdk_color_state_ref (color_state);
   bytes = NULL;
 
-  if (sigsetjmp (png_jmpbuf (png), 1))
+  if (setjmp (png_jmpbuf (png)))
     {
       gdk_color_state_unref (color_state);
       g_clear_pointer (&bytes, g_bytes_unref);
diff -Naur gtk4/meson.build gtk4-clean/meson.build
--- gtk4/meson.build	2026-04-19 02:33:20.000000000 +0300
+++ gtk4-clean/meson.build	2026-05-29 12:11:54.038340744 +0300
@@ -608,28 +608,12 @@
   cdata.set('HAVE_XDAMAGE', 1)
   cdata.set('HAVE_XFIXES', 1)
 
-  if not cc.has_function('XkbQueryExtension', dependencies: x11_dep,
-                         prefix : '#include <X11/XKBlib.h>')
-    error('X11 backend enabled, but Xkb not found.')
-  endif
   cdata.set('HAVE_XKB', 1)
 
-  if not cc.has_function('XSyncQueryExtension', dependencies: xext_dep,
-                         prefix: '''#include <X11/Xlib.h>
-                                    #include <X11/extensions/sync.h>''')
-    error('X11 backend enabled, but Xsync not found.')
-  endif
   cdata.set('HAVE_XSYNC', 1)
 
-  if not cc.has_function('XGetEventData', dependencies: x11_dep)
-    error('X11 backend enabled, but no generic event support.')
-  endif
   cdata.set('HAVE_XGENERICEVENTS', 1)
 
-  if not xi_dep.found() or not cc.has_header('X11/extensions/XInput2.h', dependencies: xi_dep)
-    error('X11 backend enabled, but XInput2 not found.')
-  endif
-
   # Note that we also check that the XIScrollClassInfo struct is defined,
   # because at least Ubuntu Oneiric seems to have XIAllowTouchEvents(),
   #   but not the XIScrollClassInfo struct
@@ -637,24 +621,11 @@
   has_scroll_class_info = cc.has_member('XIScrollClassInfo', 'number', dependencies: xi_dep,
                                         prefix: '''#include <X11/Xlib.h>
                                                    #include <X11/extensions/XInput2.h>''')
-  if not has_allow_touch_events or not has_scroll_class_info
-    error('X11 backend enabled, but XInput2.2 not found.')
-  endif
-  cdata.set('XINPUT_2_2', 1)
-
-  has_gesture_pinch_event = cc.has_member('XIGesturePinchEvent', 'type', dependencies: xi_dep,
-                                          prefix: '''#include <X11/Xlib.h>
-                                                     #include <X11/extensions/XInput2.h>''')
-  if has_gesture_pinch_event
-    cdata.set('XINPUT_2_4', 1)
-  endif
+  cdata.set('XINPUT_2_2', 0)
+  cdata.set('XINPUT_2_4', 0)
 
   xinerama_dep = dependency('xinerama')
-  if not cc.has_header_symbol('X11/extensions/Xinerama.h', 'XineramaQueryExtension', dependencies: xinerama_dep)
-    error('X11 backend enabled, but Xinerama extension does not work.')
-  endif
   cdata.set('HAVE_XFREE_XINERAMA', 1)
-
   cdata.set('HAVE_RANDR', xrandr_dep.found())
   cdata.set('HAVE_RANDR15', xrandr15_dep.found())
 endif
@@ -726,9 +697,6 @@
   cdata.set('HAVE_DRM_DRM_FOURCC_H', have_drm_fourcc)
 endif
 
-cdata.set('HAVE_DMABUF', cc.has_header('linux/dma-buf.h'))
-cdata.set('HAVE_UDMABUF', cc.has_header('linux/udmabuf.h'))
-
 cloudproviders_dep = dependency('cloudproviders',
   required: get_option('cloudproviders'),
   version: cloudproviders_req,
