diff -Naur gweather4/meson.build gweather4-clean/meson.build
--- gweather4/meson.build	2023-08-07 17:23:58.000000000 +0300
+++ gweather4-clean/meson.build	2026-08-22 22:29:04.032600862 +0300
@@ -47,7 +47,6 @@
   '-d', 'C0326',
 ]
 
-py = import('python').find_installation('python3', modules: ['gi'])
 gen_locations_variant = find_program('build-aux/meson/gen_locations_variant.py')
 
 g_ir_scanner = find_program('g-ir-scanner', required: get_option('introspection'))
