diff -Naur rust-ipc-channel/Cargo.toml rust-ipc-channel-clean/Cargo.toml
--- rust-ipc-channel/Cargo.toml	1970-01-01 03:00:01.000000000 +0300
+++ rust-ipc-channel-clean/Cargo.toml	2026-07-23 12:48:13.329695721 +0300
@@ -97,15 +97,15 @@
 memfd = ["sc"]
 win32-trace = []
 
-[target.'cfg(any(target_os = "linux", target_os = "openbsd", target_os = "freebsd"))'.dependencies.mio]
+[target.'cfg(any(target_os = "linux", target_os = "openbsd", target_os = "freebsd", target_os = "orange"))'.dependencies.mio]
 version = "1.0"
 features = ["os-ext"]
 
-[target.'cfg(any(target_os = "linux", target_os = "openbsd", target_os = "freebsd"))'.dependencies.sc]
+[target.'cfg(any(target_os = "linux", target_os = "openbsd", target_os = "freebsd", target_os = "orange"))'.dependencies.sc]
 version = "0.2.2"
 optional = true
 
-[target.'cfg(any(target_os = "linux", target_os = "openbsd", target_os = "freebsd"))'.dependencies.tempfile]
+[target.'cfg(any(target_os = "linux", target_os = "openbsd", target_os = "freebsd", target_os = "orange"))'.dependencies.tempfile]
 version = "3.4"
 
 [target.'cfg(target_os = "windows")'.dependencies.windows]
diff -Naur rust-ipc-channel/src/platform/mod.rs rust-ipc-channel-clean/src/platform/mod.rs
--- rust-ipc-channel/src/platform/mod.rs	2006-07-24 05:21:28.000000000 +0400
+++ rust-ipc-channel-clean/src/platform/mod.rs	2026-07-23 12:48:49.598876428 +0300
@@ -9,12 +9,12 @@
 
 #[cfg(all(
     not(feature = "force-inprocess"),
-    any(target_os = "linux", target_os = "openbsd", target_os = "freebsd")
+    any(target_os = "linux", target_os = "openbsd", target_os = "freebsd", target_os = "orange")
 ))]
 mod unix;
 #[cfg(all(
     not(feature = "force-inprocess"),
-    any(target_os = "linux", target_os = "openbsd", target_os = "freebsd")
+    any(target_os = "linux", target_os = "openbsd", target_os = "freebsd", target_os = "orange")
 ))]
 mod os {
     pub use super::unix::*;
diff -Naur rust-ipc-channel/src/platform/unix/mod.rs rust-ipc-channel-clean/src/platform/unix/mod.rs
--- rust-ipc-channel/src/platform/unix/mod.rs	2006-07-24 05:21:28.000000000 +0400
+++ rust-ipc-channel-clean/src/platform/unix/mod.rs	2026-07-23 13:21:10.341027267 +0300
@@ -1139,7 +1139,7 @@
                 }
             },
             BlockingMode::Timeout(duration) => {
-                #[cfg(target_os = "linux")]
+                #[cfg(target_os = "orange")]
                 const POLLRDHUP: libc::c_short = libc::POLLRDHUP;
 
                 // It's rather unfortunate that Rust's libc doesn't know that
