diff -Naur rust-cap-primitives/src/rustix/fs/dir_utils.rs rust-cap-primitives-clean/src/rustix/fs/dir_utils.rs
--- rust-cap-primitives/src/rustix/fs/dir_utils.rs	2006-07-24 05:21:28.000000000 +0400
+++ rust-cap-primitives-clean/src/rustix/fs/dir_utils.rs	2026-06-09 16:02:30.074436190 +0300
@@ -148,6 +148,7 @@
         target_os = "wasi",
         target_os = "illumos",
         target_os = "solaris",
+        target_os = "orange"
     ))]
     {
         OFlags::empty()
diff -Naur rust-cap-primitives/src/rustix/fs/oflags.rs rust-cap-primitives-clean/src/rustix/fs/oflags.rs
--- rust-cap-primitives/src/rustix/fs/oflags.rs	2006-07-24 05:21:28.000000000 +0400
+++ rust-cap-primitives-clean/src/rustix/fs/oflags.rs	2026-06-09 16:01:58.990623493 +0300
@@ -31,7 +31,8 @@
         target_os = "watchos",
         target_os = "visionos",
         target_os = "freebsd",
-        target_os = "fuchsia"
+        target_os = "fuchsia",
+        target_os = "orange"
     )))]
     if options.rsync {
         oflags |= OFlags::RSYNC;
