diff -Naur rust-mio1.0.1/src/sys/unix/mod.rs rust-mio1.0.1-clean/src/sys/unix/mod.rs
--- rust-mio1.0.1/src/sys/unix/mod.rs	2024-07-23 14:26:56.000000000 +0300
+++ rust-mio1.0.1-clean/src/sys/unix/mod.rs	2026-08-01 15:39:17.849939366 +0300
@@ -22,6 +22,7 @@
             target_os = "illumos",
             target_os = "linux",
             target_os = "redox",
+            target_os = "orange"
         )
     ), path = "selector/epoll.rs")]
     #[cfg_attr(all(
@@ -100,6 +101,7 @@
         target_os = "redox",
         target_os = "solaris",
         target_os = "vita",
+        target_os = "orange"
     ), path = "waker/pipe.rs")]
     mod waker;
     // NOTE: the `Waker` type is expected in the selector module as the
diff -Naur rust-mio1.0.1/src/sys/unix/pipe.rs rust-mio1.0.1-clean/src/sys/unix/pipe.rs
--- rust-mio1.0.1/src/sys/unix/pipe.rs	2024-07-23 14:26:56.000000000 +0300
+++ rust-mio1.0.1-clean/src/sys/unix/pipe.rs	2026-08-01 15:38:58.277637364 +0300
@@ -21,6 +21,7 @@
         target_os = "redox",
         target_os = "solaris",
         target_os = "vita",
+        target_os = "orange"
     ))]
     unsafe {
         if libc::pipe2(fds.as_mut_ptr(), libc::O_CLOEXEC | libc::O_NONBLOCK) != 0 {
diff -Naur rust-mio1.0.1/src/sys/unix/tcp.rs rust-mio1.0.1-clean/src/sys/unix/tcp.rs
--- rust-mio1.0.1/src/sys/unix/tcp.rs	2024-07-23 14:26:56.000000000 +0300
+++ rust-mio1.0.1-clean/src/sys/unix/tcp.rs	2026-08-01 15:39:04.169473852 +0300
@@ -102,6 +102,7 @@
         target_os = "vita",
         target_os = "hermit",
         target_os = "nto",
+        target_os = "orange",
         all(target_arch = "x86", target_os = "android"),
     ))]
     let stream = {
