diff -Naur libgpg-error-1.55/build-aux/config.sub libgpg-error-1.55-patched/build-aux/config.sub
--- libgpg-error-1.55/build-aux/config.sub	2022-08-02 15:53:50.000000000 +0300
+++ libgpg-error-1.55-patched/build-aux/config.sub	2025-09-25 13:58:17.464414286 +0300
@@ -1750,7 +1750,7 @@
 	     | os2* | vos* | palmos* | uclinux* | nucleus* | morphos* \
 	     | scout* | superux* | sysv* | rtmk* | tpf* | windiss* \
 	     | powermax* | dnix* | nx6 | nx7 | sei* | dragonfly* \
-	     | skyos* | haiku* | rdos* | toppers* | drops* | es* \
+	     | skyos* | orange* | haiku* | rdos* | toppers* | drops* | es* \
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
 	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx* | zephyr* \
diff -Naur libgpg-error-1.55/src/syscfg/lock-obj-pub.orange.h libgpg-error-1.55-patched/src/syscfg/lock-obj-pub.orange.h
--- libgpg-error-1.55/src/syscfg/lock-obj-pub.orange.h	1970-01-01 03:00:00.000000000 +0300
+++ libgpg-error-1.55-patched/src/syscfg/lock-obj-pub.orange.h	2018-10-12 18:12:38.000000000 +0300
@@ -0,0 +1,25 @@
+## lock-obj-pub.x86_64-pc-linux-gnu.h
+## File created by gen-posix-lock-obj - DO NOT EDIT
+## To be included by mkheader into gpg-error.h
+
+typedef struct
+{
+  long _vers;
+  union {
+    volatile char _priv[40];
+    long _x_align;
+    long *_xp_align;
+  } u;
+} gpgrt_lock_t;
+
+#define GPGRT_LOCK_INITIALIZER {1,{{0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0}}}
+##
+## Local Variables:
+## mode: c
+## buffer-read-only: t
+## End:
+##
