diff -Naur ncurses-6.5/config.sub ncurses-patched/config.sub
--- ncurses-6.5/config.sub	2023-12-27 17:41:27.000000000 +0300
+++ ncurses-patched/config.sub	2025-06-02 08:30:55.408408771 +0300
@@ -1764,7 +1764,7 @@
 	     | os2* | vos* | palmos* | uclinux* | nucleus* | morphos* \
 	     | scout* | superux* | sysv* | rtmk* | tpf* | windiss* \
 	     | powermax* | dnix* | nx6 | nx7 | sei* | dragonfly* \
-	     | skyos* | haiku* | rdos* | toppers* | drops* | es* \
+	     | skyos* | orange* | haiku* | rdos* | toppers* | drops* | es* \
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
 	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx* | zephyr* \
diff -Naur ncurses-6.5/configure ncurses-patched/configure
--- ncurses-6.5/configure	2024-04-10 11:09:48.000000000 +0300
+++ ncurses-patched/configure	2025-06-02 09:00:16.635459472 +0300
@@ -6925,6 +6925,10 @@
 		LINK_PROGS="$SHELL ${rel_builddir}/mk_prog.sh"
 		LINK_TESTS="$SHELL ${rel_builddir}/mk_prog.sh"
 		;;
+	(orange*)
+		CC_SHARED_OPTS='-fPIC'
+		MK_SHARED_LIB='${CC} -shared -o $@'
+		;;
 	(mingw*)
 		cf_cv_shlib_version=mingw
 		cf_cv_shlib_version_infix=mingw