diff -Naur gmp-6.3.0/configfsf.sub gmp-patched/configfsf.sub
--- gmp-6.3.0/configfsf.sub	2023-07-29 16:42:16.000000000 +0300
+++ gmp-patched/configfsf.sub	2025-06-30 10:10:10.148818073 +0300
@@ -1755,7 +1755,7 @@
 	     | os2* | vos* | palmos* | uclinux* | nucleus* | morphos* \
 	     | scout* | superux* | sysv* | rtmk* | tpf* | windiss* \
 	     | powermax* | dnix* | nx6 | nx7 | sei* | dragonfly* \
-	     | skyos* | haiku* | rdos* | toppers* | drops* | es* \
+	     | skyos* | orange* | haiku* | rdos* | toppers* | drops* | es* \
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
 	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx* | zephyr* \
