diff -Naur mpfr-4.2.2/config.sub mpfr-patched/config.sub
--- mpfr-4.2.2/config.sub	2025-03-20 12:21:11.000000000 +0300
+++ mpfr-patched/config.sub	2025-06-30 11:27:44.186453150 +0300
@@ -1750,7 +1750,7 @@
 	     | os2* | vos* | palmos* | uclinux* | nucleus* | morphos* \
 	     | scout* | superux* | sysv* | rtmk* | tpf* | windiss* \
 	     | powermax* | dnix* | nx6 | nx7 | sei* | dragonfly* \
-	     | skyos* | haiku* | rdos* | toppers* | drops* | es* \
+  	     | skyos* | orange* | haiku* | rdos* | toppers* | drops* | es* \
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
 	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx* | zephyr* \
