diff -Naur findutils-4.6.0/build-aux/config.sub findutils-patched/build-aux/config.sub
--- findutils-4.6.0/build-aux/config.sub	2015-12-22 20:32:53.000000000 +0300
+++ findutils-patched/build-aux/config.sub	2025-06-17 07:39:31.339151302 +0300
@@ -1398,7 +1398,7 @@
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
-	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* \
+	      | -skyos* | -orange* | -haiku* | -rdos* | -toppers* | -drops* | -es* \
 	      | -onefs* | -tirtos*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
